search-commons/src/main/java/no/unit/nva/search/IndexDocumentGenerator.java

```diff
@@ -214,6 +240,47 @@ private static Instant extractPublishedDate(JsonNode record, UUID id) {
         return getInstant(record, id, PUBLISHED_DATE_JSON_POINTER, PUBLISHED_DATE);
     }
 
+    private static Map<String, String> extractAlternativeTitles(JsonNode record, UUID id) {
+        JsonNode node = record.at(ALTERNATIVETITLES_JSON_POINTER);
+        String json = node.toString();
+        Map<String, String> map = Collections.EMPTY_MAP;
+        try {
+            Map<String, AttributeValue> attributeValueMap = mapper.readValue(json, MAP_ATTRIBUTEVALUE_JAVA_TYPE);
```

Comment:
json, replace with record.at(ALTERNATIVETITLES_JSON_POINTER).toString()

Reply:
In fact, extract a method that does this and takes the pointer as a value. Use that everywhere